framework/src/utils/RankFourTensor.C

```diff
@@ -831,3 +831,52 @@ RankFourTensor::sum3x1() const
   a(2) = _vals[2][2][0][0] + _vals[2][2][1][1] + _vals[2][2][2][2]; // C2200 + C2211 + C2222
   return a;
 }
+
+bool
+RankFourTensor::isSymmetric() const
+{
+  for (unsigned int i = 1; i < N; ++i)
+    for (unsigned int j = 0; j < i; ++j)
+      for (unsigned int k = 1; k < N; ++k)
+        for (unsigned int l = 0; l < k; ++l)
+          if (_vals[i][j][k][l] != _vals[j][i][k][l] || _vals[i][j][k][l] != _vals[i][j][l][k])
+            return false;
+  return true;
+}
+
+bool
+RankFourTensor::isIsotropic() const
+{
+  // prerequisite is symmetry
+  if (!isSymmetric())
+    return false;
+
+  // inspect shear components
+  const Real mu = _vals[0][1][0][1];
+  if (_vals[1][2][1][2] != mu || _vals[2][0][2][0] != mu)
+    return false;
+  if (_vals[2][0][1][2] != 0.0 || _vals[0][1][1][2] != 0.0 || _vals[0][1][2][0] != 0.0 ||
+      _vals[1][2][2][0] != 0.0 || _vals[1][2][0][1] != 0.0 || _vals[2][0][0][1] != 0.0)
+    return false;
+
+  // off diagonal blocks in Voigt
+  for (unsigned int i = 0; i < N; ++i)
+    for (unsigned int j = 0; j < N; ++j)
+      if (_vals[i][i][(j + 1) % N][(j + 2) % N] != 0.0 ||
+          _vals[(j + 1) % N][(j + 2) % N][i][i] != 0.0)
+        return false;
+
+  // top left block
+  const Real K1 = _vals[0][0][0][0];
+  const Real K2 = _vals[0][0][1][1];
+  if (!MooseUtils::absoluteFuzzyEqual(K1 - 4.0 * mu / 3.0, K2 + 2.0 * mu / 3.0))
+    return false;
+  if (_vals[1][1][1][1] != K1 || _vals[2][2][2][2] != K1)
```

Comment:
Do you think any of these other equality checks should be fuzzied up?  I guess since they're usually copied from the same number, that might not be necessary.  How about the ones where you're comparing with zero?  They're directly set to zero, so maybe it's OK.